panther_analysis_tool/main.py

```diff
@@ -419,16 +419,30 @@ def run_tests(analysis: Dict[str, Any], analysis_funcs: Dict[str, Any],
             failed_tests[analysis.get('PolicyID') or
                          analysis['RuleID']].append(unit_test['Name'])
             continue
-        test_result = 'PASS'
+
+        test_result = defaultdict(lambda: 'PASS')
+        # check expected result
         if result != unit_test['ExpectedResult']:
-            test_result = 'FAIL'
+            test_result['expected'] = 'FAIL'
+
+        # check dedup and title function return non-None
+        if unit_test['ExpectedResult']:
```

Comment:
What's the purpose of this conditional? Could the expected result be None for a unit test? I wasn't aware of that.

Or do we only care about the dedup/title when the test case is supposed to return True? Keep in mind rules and policies have flipped logic, so a normal/compliant policy is True, but a normal/unmatched rule is False.

I suspect we could remove this line, but I'm not 100% sure

Reply:
We only care about the dedup/title when the test case is supposed to be True. You are correct about rules & policies having flipped logic, but this is one of those weird peculiarities between rules & policies because rules are the only ones that support title/dedup functions. Policies don't support these, and the caller function of this function handles that by not passing in a title/dedup function in `analysis_funcs` for policies.

Reply:
Ah, maybe a comment to that effect then:

```go
# Only applies to rules which match an incoming event
```

Reply:
Added